src/libs/actions/Report.ts

```diff
@@ -2164,7 +2171,17 @@ function leaveRoom(reportID: string, isWorkspaceMemberLeavingWorkspaceRoom = fal
 
     API.write('LeaveRoom', parameters, {optimisticData, successData, failureData});
 
-    if (isWorkspaceMemberLeavingWorkspaceRoom) {
+    const lastAccessedReportID = ReportUtils.sortReportsByLastRead(allReports, reportMetadata)
+        .filter(
+            (r) =>
+                r?.reportID !== reportID && r?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN && ReportUtils.shouldReportBeInOptionList(r, '', false, [], {}, true),
+        )
+        .at(-1)?.reportID;
```

Comment:
can we change this to something like this for clear readability?

```js
    const sortedReportsByLastRead = ReportUtils.sortReportsByLastRead(allReports, reportMetadata);
    // lets add some comment here
    const filteredReportsByLastRead = sortedReportsByLastRead.filter((report) => (
        report?.reportID !== reportID &&
        report?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN &&
        ReportUtils.shouldReportBeInOptionList(report, '', false, [], {}, true)
    ));
    const lastAccessedReportID = filteredReportsByLastRead.at(-1)?.reportID;
```

Reply:
and add one empty line before the if statement?

Reply:
@getusha 

Hmm, `report` is already defined in the scope which fails `npm run lint` and the `npm run prettier` moves filter conditions to a single line. The formatted code will like

```js
    const sortedReportsByLastRead = ReportUtils.sortReportsByLastRead(allReports, reportMetadata);
    // We want to filter out the current report, hidden reports and empty chats
    const filteredReportsByLastRead = sortedReportsByLastRead.filter(
        (r) => r?.reportID !== reportID && r?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN && ReportUtils.shouldReportBeInOptionList(r, '', false, [], {}, true),
    );
    const lastAccessedReportID = filteredReportsByLastRead.at(-1)?.reportID;
```

Does it look good to you? Or other suggestions about formatting the code?

Reply:
Looks good 👍